### PR TITLE
feat(winget): support installing .exe directly

### LINKED ISSUE #4494: Publishing to winget without archiving
### Is your feature request related to a problem? Please describe.

Winget support in GoReleaser fails when `archive: [ format:binary]` is set.

### Describe the solution you'd like

Instead,  GoReleaser should create manifests, with a direct URL to the binary, replacing `InstallerType: zip` with `InstallerType: portable` and omitting NestedInstaller fields.

### Describe alternatives you've considered

N/A

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by caarlos0**
closes #4494

### CODE REVIEW

**Review by vedantmgoyal2009 — APPROVED** [review]

File: internal/pipe/winget/winget.go [hunk 1]
```diff
@@ -231,32 +237,32 @@ func (p Pipe) doRun(ctx *context.Context, winget config.Winget, cl client.Releas
...
+			UpgradeBehavior: "uninstallPrevious",
+		}
+		if archive.Format() == "zip" {
+			zipCount++
+			installer.InstallerType = "zip"
+			item.NestedInstallerType = "portable"
+			item.NestedInstallerFiles = installerItemFilesFor(*archive)
+		} else {
+			binaryCount++
+			installer.InstallerType = "portable"
```
Comment (vedantmgoyal2009) [inline internal/pipe/winget/winget.go:263]:
```suggestion
			installer.InstallerType = "portable"
			installer.Commands = []string{winget.Name}
```

**Review by vedantmgoyal2009 — CHANGES_REQUESTED** [review]

**Review by vedantmgoyal2009 — CHANGES_REQUESTED** [review]
Package identifier is `Publisher.Project`, so artifacts should start with `min` (project) and not `foo` (publisher).

**Comment by vedantmgoyal2009** [comment]
well @caarlos0, we can change the package identifier to `Min.Foo`, which I think will be easier than replacing the artifact names. however, the moniker also needs to be updated to `foo` and also `PackageName` and `Publisher` have to be inter-changed in Locale manfiest.

**Comment by caarlos0** [comment]
updated

**Comment by caarlos0** [comment]
@vedantmgoyal2009 updated

**Pull request merged**